### PR TITLE
ITE-3: Real-world example of combining TUF and in-toto for packaging Datadog Agent integrations

### PR DESCRIPTION
**Pull request by trishankatdatadog**
Datadog is a monitoring service for cloud-scale applications that monitors servers, databases, tools, and services through a software-as-a-service-based data analytics platform. It supports multiple cloud service providers, including Amazon Web Services, Microsoft Azure, Google Cloud Platform, and Red Hat OpenShift. At the time of this writing, the company servers more than 8,000 customers, and collects trillions of monitoring record points on a daily basis.

The Datadog agent is the software that runs on virtual machines or containers. It collects events and metrics from these virtual machines or containers and sends them to Datadog, where customers can analyze their monitoring and performance data. The agent integrations are plug-ins that collect metrics from services running on customer infrastructure. Presently, there are more than one hundred integrations that come installed out-of-the-box with the Agent.

This ITE discusses the TUF security model used to distribute the Datadog Agent integrations in a compromise-resilient manner.

### CODE REVIEW

**Comment by trishankatdatadog** [comment]
@jhdalek55 @SantiagoTorres @JustinCappos Please send feedback, thanks!

File: ITE/3/README.adoc [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+|===
+
+[[abstract]]
+== Abstract
+
+Datadog is a monitoring service for cloud-scale applications that monitors
+servers, databases, tools, and services through a
+software-as-a-service-based data analytics platform. It supports
+multiple cloud service providers, including Amazon Web Services,
+Microsoft Azure, Google Cloud Platform, and Red Hat OpenShift. At the
```
Comment (JustinCappos) [inline ITE/3/README.adoc]: This last sentence feels a little too much like an advertisement and will likely change over time (hopefully growing!).  Would it be okay to remove?

File: ITE/3/README.adoc [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+presently has three steps:
+
+[arabic]
+. The first "`tag`" step outputs Python source code as products. Every
+integration consists of Python source code and several
+https://yaml.org/[YAML] configuration files. The step uses (1, n)
+Yubikeys, where n is the current number of Datadog Agent integration
+developers. (See this
+https://dd-integrations-core-wheels-build-stable.datadoghq.com/targets/in-toto-metadata/3e592d47ab815cb9bd8aa064bbaa2a217c1baa9a55a7f8b73d2da6dfd14a247e/tag.c295cf63.link[example]
+"`tag`" link metadata file.)
```
Comment (JustinCappos) [inline ITE/3/README.adoc]: Can you give a little context of why you made these choices?  What you did is valuable to know.  Why you did it is even better!

File: ITE/3/README.adoc [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+the root metadata file.) All TUF metadata and targets are written using
+https://github.com/theupdateframework/specification/blob/master/tuf-spec.md#7-consistent-snapshots[consistent
+snapshots]. Figure 3 summarizes the security model discussed in this
+subsection.
+
+image:tuf-in-toto-figure-3.png[Figure 3]
+
+*Figure 3*: A summary of the security model discussed in this subsection.
+
+In summary, the Datadog pipeline uses TUF to appropriately bootstrap the
```
Comment (JustinCappos) [inline ITE/3/README.adoc]:
```suggestion
In summary, the Datadog pipeline uses TUF to bootstrap the
```
Suggested removal because this feels subjective.

**Comment by trishankatdatadog** [comment]
@JustinCappos Thanks! Does the security analysis make sense to you?

**Comment by JustinCappos** [comment]
Sort of.  I wonder about multi-key / role attacks, especially those that
might be likely given your deployment model.

I like the general concept though.

**Comment by trishankatdatadog** [comment]
Thanks for all your help, @adityasaky!

@JustinCappos, could we please get another review?

**Comment by trishankatdatadog** [comment]
@JustinCappos Okay, I resolved your comments above. Is there anything else you'd like to see, or we can we merge now?

**Comment by trishankatdatadog** [comment]
Thanks! Could we merge this?

**Comment by SantiagoTorres** [comment]
LGTM, thanks!

**Pull request merged**